lib/ansible/modules/network/netscaler/netscaler_save_config.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#  Copyright (c) 2017 Citrix Systems
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: netscaler_save_config
+short_description: Save Netscaler configuration.
+description:
+    - This module uncoditionally saves the configuration on the target netscaler node.
+    - This module does not support check mode.
+    - This module is intended to run either on the ansible  control node or a bastion (jumpserver) with access to the actual netscaler instance.
+
+version_added: "2.4.0"
+
+author: George Nikolopoulos (@giorgos-nikolopoulos)
+
+options:
+    nsip:
+        description:
+            - The ip address of the netscaler appliance where the nitro API calls will be made.
+            - "The port can be specified with the colon (:). E.g. C(192.168.1.1:555)."
+        required: True
+
+    nitro_user:
+        description:
+            - The username with which to authenticate to the netscaler node.
+        required: True
+
+    nitro_pass:
+        description:
+            - The password with which to authenticate to the netscaler node.
+        required: True
+
+    nitro_protocol:
+        choices: [ 'http', 'https' ]
+        default: http
+        description:
+            - Which protocol to use when accessing the nitro API objects.
+
+    validate_certs:
+        description:
+            - If C(no), SSL certificates will not be validated. This should only be used on personally controlled sites using self-signed certificates.
+        required: false
+        default: 'yes'
+
+    nitro_timeout:
+        description:
+            - Time in seconds until a timeout error is thrown when establishing a new session with Netscaler.
+        default: 310
+
+requirements:
+    - nitro python sdk
+'''
+
+EXAMPLES = '''
+---
+- name: Save netscaler configuration
+  delegate_to: localhost
+  netscaler_save_config:
+    nsip: 172.18.0.2
+    nitro_user: nsroot
+    nitro_pass: nsroot
+
+- name: Setup server without saving  configuration
+  delegate_to: localhost
```

Comment:
The modules make http requests utilizing the NITRO API.

The NITRO API allows access from outside the Netscaler node. If you try to run the http requests form inside Netscaler to locahost they will fail.

So the ``delegate_to`` semantically means go to the node that has http access to the Netscaler node and run the http requests there.

In our examples we use only the local machine but in real world deployments it is common to have "bastion" hosts that are used to access the actual Netscaler.

In such a case we would utilize ``delegate_to: bastion_host``

As far as the ``connection: local`` I am not very familiar with how it is used.

Reading the documentation I get the impression that semantically it means that the actual node to configure is the ansible node. Also I don't know how you could emulate the bastion server use case.

In the inline example the values for ``nsip``, ``nitro_user``, ``nitro_pass`` are given directly while in most real world cases they would take values from varibales.

In such case, you can run the example with ``connection:local`` but you would have to change ``hosts: 127.0.0.1`` and also provide values for the variables ``nsip``, ``nitro_user``, ``nitro_pass`` which in the [samples](https://github.com/citrix/netscaler-ansible-modules/tree/master/samples) we have in our repository are defined as per host variables for the netscaler node.

It seems to me that ``delegate_to`` is more fitting for the use cases we have for Netscaler.